From e97dcef7bb3fa143181a68b36eaa3fb02c814d32 Mon Sep 17 00:00:00 2001
From: monsta <monsta@inbox.ru>
Date: Thu, 5 Mar 2015 17:21:31 +0300
Subject: [PATCH] ev-jobs: don't forget to unlock the trylock'ed mutex

---
 libview/ev-jobs.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/libview/ev-jobs.c b/libview/ev-jobs.c
index d8d3043..99cfc8c 100644
--- a/libview/ev-jobs.c
+++ b/libview/ev-jobs.c
@@ -834,9 +834,11 @@ ev_job_fonts_run (EvJob *job)
 	/* Do not block the main loop */
 	if (!ev_document_doc_mutex_trylock ())
 		return TRUE;
-	
-	if (!ev_document_fc_mutex_trylock ())
+
+	if (!ev_document_fc_mutex_trylock ()) {
+		ev_document_doc_mutex_unlock ();
 		return TRUE;
+	}
 
 #ifdef EV_ENABLE_DEBUG
 	/* We use the #ifdef in this case because of the if */

