Description: Add keywords key to .desktop files
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Forwarded: https://github.com/mate-desktop/atril/issues/78 
--- a/data/atril.desktop.in.in
+++ b/data/atril.desktop.in.in
@@ -14,4 +14,5 @@
 X-MATE-Bugzilla-Component=BugBuddyBugs
 X-MATE-Bugzilla-Version=@VERSION@
 Categories=GTK;Office;Viewer;
+Keywords=MATE;document;viewer;pdf;dvi;ps;xps;tiff;pixbuf;djvu;comics;
 MimeType=@ATRIL_MIME_TYPES@
